src/order/prime_ideal.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2021 Noam Atar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Noam Atar
+-/
+import order.basic
+import order.ideal
+import order.pfilter
+
+/-!
+# Prime ideals
+
+## Main definitions
+
+Throughout this file, `P` is at least a preorder, but some sections require more
+structure, such as a bottom element, a top element, or a join-semilattice structure.
+
+- `prime_pair`: A pair of an `ideal` and a `pfilter` who form a partition of `P`.
+                This is useful as giving the data of a prime ideal is the same
+                as giving the data of a prime pfilter.
+- `is_prime`: a predicate for prime ideals.
+              Dual to the notion of a prime filter.
+
+## References
+
+- <https://en.wikipedia.org/wiki/Ideal_(order_theory)>
+
+## Tags
+
+ideal, prime
+
+-/
+
+namespace order
+
+variables {P : Type*}
+
+namespace ideal
+
+section preorder
+
+variables [preorder P]
+
+/-- A pair of an `ideal` and a `pfilter` who form a partition of `P`.
+-/
+structure prime_pair (P : Type*) [preorder P] :=
+(I        : ideal P)
+(F        : pfilter P)
+(disjoint : (I : set P) ∩ F = ∅)
+(cover    : (I : set P) ∪ F = set.univ)
+
+lemma prime_pair.ideal_compl_eq_pfilter {IF : prime_pair P} : (IF.I : set P)ᶜ = IF.F :=
```

Comment:
I like your choice of `IF` as the name of a generic `prime_pair`, it's nicely mnemonic.